Police Strobe no output when main channel has lowest output saved

I can't get the police strobe blink mode to function correctly. It seems like there might be a glitch in the mode memory.

Using the latest trunk

If the light is set to moonlight ( 1/150 ) for the main channel, switching to the police strobe results in no output. 1H doesn't do anything in this mode.

Turning the light off, back on to main channel and setting it above 1/150, then going back to the strobe there's output as expected.

Seems like it's just an issue with the lowest mode.

I have this issue with the TS10-rgbaux, TS11 and TS25.

I thought I was doing something wrong for a few hours till I figured out it was the output setting that was causing the issue.

The firmware that shipped with my TS10 didn't have this issue, though I am not certain what version it was. I would have to flash it back and check it. The TS11 and TS25 both shipped with firmware versions that didn't have the police strobe.

Seems very strange as the police strobe doesn't seem to even have a brightness setting. So I have no idea why it being on moonlight for the main channel would result in no output on the aux.

Ah well that checks out then. Thank you.

Still seems like a bug as I thought the brightness wasn't an option for police strobe ( e.g. shouldn't it always be aux high? )

Trying to use that PRs changes I get a compiler error. I also tried to do the same thing myself ( though slightly different, not with the bitwise operations ) and am hitting the same error. I likely am just not familiar enough with the project, missing an include somewhere or something silly. Or it's probably just out of date with the main branch, but I'm not sure what all changed or where to start or look. Any help is always greatly appreciated.

```
In file included from hw/wurkkos/ts25/hwdef.c:7:0,
                 from ui/anduril/anduril.c:88:
./fsm/chan-rgbaux.c:6:1: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘void’
 void set_level_auxred(uint8_t level) {
 ^
```

--- a/fsm/chan-aux.c
+++ b/fsm/chan-aux.c
@@ -4,7 +4,7 @@
 #pragma once
 
 void set_level_aux(uint8_t level) {
-    indicator_led(!(!(level)) << 1);  // high (or off)
+    indicator_led((!(!(level)) << 1) + 1);  // high (level > 0) or low
 }
 
 bool gradual_tick_null(uint8_t gt) { return true; }  // do nothing

--- a/fsm/chan-rgbaux.c
+++ b/fsm/chan-rgbaux.c
@@ -4,31 +4,31 @@
 #pragma once
 
 void set_level_auxred(uint8_t level) {
-    rgb_led_set(!(!(level)) * 0b000010);  // red, high (or off)
+    rgb_led_set(0b000001 << !(!(level)));  // red, high (level > 0) or low
 }
 
 void set_level_auxyel(uint8_t level) {
-    rgb_led_set(!(!(level)) * 0b001010);  // red+green, high (or off)
+    rgb_led_set(0b000101 << !(!(level)));  // red+green, high (level > 0) or low
 }
 
 void set_level_auxgrn(uint8_t level) {
-    rgb_led_set(!(!(level)) * 0b001000);  // green, high (or off)
+    rgb_led_set(0b000100 << !(!(level)));  // green, high (level > 0) or low
 }
 
 void set_level_auxcyn(uint8_t level) {
-    rgb_led_set(!(!(level)) * 0b101000);  // green+blue, high (or off)
+    rgb_led_set(0b010100 << !(!(level)));  // green+blue, high (level > 0) or low
 }
 
 void set_level_auxblu(uint8_t level) {
-    rgb_led_set(!(!(level)) * 0b100000);  // blue, high (or off)
+    rgb_led_set(0b010000 << !(!(level)));  // blue, high (level > 0) or low
 }
 
 void set_level_auxprp(uint8_t level) {
-    rgb_led_set(!(!(level)) * 0b100010);  // red+blue, high (or off)
+    rgb_led_set(0b010001 << !(!(level)));  // red+blue, high (level > 0) or low
 }
 
 void set_level_auxwht(uint8_t level) {
-    rgb_led_set(!(!(level)) * 0b101010);  // red+green+blue, high (or off)
+    rgb_led_set(0b010101 << !(!(level)));  // red+green+blue, high (level > 0) or low
 }
 
 bool gradual_tick_null(uint8_t gt) { return true; }  // do nothing

